acados/ocp_nlp/ocp_nlp_common.h

```diff
@@ -93,69 +85,84 @@ typedef struct
 	struct blasfeo_dmat *W;
     struct blasfeo_dvec *y_ref;
 	int *nls_mask; // nonlinear least squares mask
-	int memsize;
 } ocp_nlp_cost_ls;
 
-//
-int ocp_nlp_cost_ls_calculate_size(ocp_nlp_cost_ls_dims *dims);
-//
-ocp_nlp_cost_ls *ocp_nlp_cost_ls_assign(ocp_nlp_cost_ls_dims *dims, void *raw_memory);
-
 
 
 /************************************************
-* model
+* dynamics
 ************************************************/
 
-/* explicit ODEs */
+/* ERK */
 
 typedef struct
 {
-	ocp_nlp_dims *dims; // TODO model dims ???
-    casadi_function_t *vde;
-    casadi_function_t *vde_adj;
-    casadi_function_t *jac;
-	int memsize;
-} ocp_nlp_model_expl;
+	ocp_nlp_dims *dims; // TODO dynamics dims ???
+	external_function_generic **forw_vde;
+	external_function_generic **adj_vde;
+	external_function_generic **jac_ode;
+} ocp_nlp_dynamics_erk;
```

Comment:
See comment on the ``ocp_nlp_cost_ls``-struct. This definition shouldn't be in ``ocp_nlp_common.h``.

Reply:
Yes, the only reason why it is there atm is because @giaf did not touch the integrators yet. But eventually this will be moved inside each integrator.

Reply:
OK, I'm curious how that would look like exactly.